src/Core/Domain/Product/QueryResult/EditableProduct.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
+
+/**
+ * DTO for product that needs to be edited
+ */
+class EditableProduct
+{
+    /**
+     * @var int
+     */
+    private $productId;
+
+    /**
+     * @var string[]
+     */
+    private $localizedNames;
+
+    /**
+     * @var ProductType
+     */
+    private $type;
+
+    public function __construct(
+        int $productId,
+        array $localizedNames,
+        ProductType $type
+    ) {
+        $this->productId = $productId;
+        $this->localizedNames = $localizedNames;
+        $this->type = $type;
+    }
+
+    /**
+     * @return int
+     */
+    public function getProductId(): int
+    {
+        return $this->productId;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getLocalizedNames(): array
```

Comment:
This part is not used/tested yet, but it should be used in the behat tests when we check for localized fields (to be done in future PRs)